packages/provider/wrapper.js

```diff
@@ -19,15 +19,19 @@ module.exports = {
     // to see what web3 is sending and receiving.
     options.verbose = options.verbose || options.verboseRpc || false;
 
-    /* create wrapper functions for before/after send */
-    var preHook = this.preHook(options);
-    var postHook = this.postHook(options);
+    /* create wrapper functions for before/after send/sendAsync */
+    const preHook = this.preHook(options);
+    const postHook = this.postHook(options);
 
-    var originalSend = provider.send.bind(provider);
+    const originalSend = provider.send.bind(provider);
 
     /* overwrite method */
     provider.send = this.send(originalSend, preHook, postHook);
-
+    // path sendAsync when sendAsync is used.
+    if (provider.sendAsync) {
```

Comment:
just a small style nit: it might make this a little more clear if you assign `originalSendAsync` right next to `originalSend` and then assign `provider.sendAsync` right next to `provider.send`. this is not something I feel strongly about so feel free to ignore me

Reply:
I tried this. If I used the same pattern on line 29, it would break when using the development network like ganache, since the `provider.sendAsync` method wouldn't be available.  So `truffle console --verbose-rpc` would break.

Reply:
Include it in an `if` so it won't like you've done.